### DISCUSSION TITLE
delete an entry from resources section in kustomization yaml

### DISCUSSION THREAD

**Discussion by venki231blr** [mikefarah/yq#1949] [General]
Hello 👋🏼 ,

I'm trying to process a k8s kustomization.yaml and remove a matching entry completely. I could not find a native way using the kustomize tools so I figured I give yq a spin.

The part I'm interesting looks like this:

resources:
 - resource1.yaml
 - resource2.yaml
What I would like to do is remove entry in resources matching resource2, so that the document afterwards looks like this:

resources:
 - resource1.yaml
I thought I figure out how to only match one entry and then expand the expression to call del() on it, but I fail already the part here.

Some things I tried:

yq eval 'del(.resources.[] | select (. == "- resource2.yaml"))' kustomization.yaml => seems to be not working
😞
I'm somewhat lost here, please help me

Thank you 🙏🏼

**Comment by mikefarah** [comment]
So close, just drop the "- " as that's not part of the value, it's just yaml notation for an array entry

```
yq 'del(.resources.[] | select (. == "resource2.yaml"))' kustomization.yaml
```

**Comment by venki231blr** [comment]
Thanks Mike! Appreciate the help. 👍
